spring-cloud-gcp-samples/spring-cloud-gcp-pubsub-binder-sample/src/main/java/com/example/SinkExample.java

```diff
@@ -45,8 +47,10 @@ public void handleMessage(UserMessage userMessage) {
 				" at " + userMessage.getCreatedAt());
 	}
 
-	@StreamListener("errorChannel")
-	public void handleErrorMessage(Message<MessagingException> message) {
+	// Note that the error inputChannel is formatted as [Pub/Sub subscription name with group].[group name].errors
+	// If you change the topic name in application.properties, you also have to change the inputChannel below.
+	@ServiceActivator(inputChannel = "my-topic.my-group.my-group.errors")
```

Comment:
Why `my-group` is twice here ?

Reply:
Looking into it deeper, in our code when we see the user configure a group with a pub/sub topic, we will name the destination `topic_name + '.' + group_name`, so that is why it gets duplicated. [Code here.](https://github.com/spring-cloud/spring-cloud-gcp/blob/72d746486fedd19b16ead3cdbbcfb48c3e90c3a9/spring-cloud-gcp-pubsub-stream-binder/src/main/java/org/springframework/cloud/gcp/stream/binder/pubsub/provisioning/PubSubChannelProvisioner.java#L78)

So then when the error channel is named `destination` + `group` + `error` = `topic.group.group.error`.

But there might be a deeper issue with that `myGroup.myGroup` thing. Maybe naming convention is messing up the bean shutdown stuff.
Because now I get this issue and the application hangs a little bit when the application terminates.

```
14:14:12.617 [main] WARN  o.s.c.s.DefaultLifecycleProcessor - Failed to stop bean 'inputBindingLifecycle'
org.springframework.beans.factory.NoSuchBeanDefinitionException: No bean named 'my-topic.my-group.my-group.errors' available
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.removeBeanDefinition(DefaultListableBeanFactory.java:988)
	at org.springframework.context.support.GenericApplicationContext.removeBeanDefinition(GenericApplicationContext.java:328)
	at org.springframework.cloud.stream.binder.AbstractMessageChannelBinder.destroyBean(AbstractMessageChannelBinder.java:796)
	at org.springframework.cloud.stream.binder.AbstractMessageChannelBinder.destroyErrorInfrastructure(AbstractMessageChannelBinder.java:783)
```

Reply:
OK. And how does it work when you remove one of those `my-group` from the `inputChannel` name ?

Reply:
The error handler code does not get called.